src/hotspot/share/services/mallocHeader.inline.hpp

```diff
@@ -89,11 +75,37 @@ inline bool MallocHeader::check_block_integrity(char* msg, size_t msglen, addres
   // we test the smallest alignment we know.
   // Should we ever start using std::max_align_t, this would be one place to
   // fix up.
-  if (!is_aligned(this, sizeof(uint64_t))) {
-    *p_corruption = (address)this;
+  if (!is_aligned(header, sizeof(uint64_t))) {
+    *p_corruption = (address)header;
     jio_snprintf(msg, msglen, "block address is unaligned");
     return false;
   }
+  return true;
+}
+
+inline MallocHeader* MallocHeader::assert_block_integrity(const void* memblock) {
+  char msg[256];
+  address corruption = NULL;
```

Comment:
Note that it is not necessary to test the supposed malloc header pointer for validity. You can also do the test on the memblock itself. Since that one must always be malloc-aligned, and the malloc header is 16 bytes, it does not matter on which pointer you do the test.